### PR TITLE
Guide: Security-Focused Guide for AI Code Assistant Instructions

### PR DESCRIPTION
**Pull request by balteravishay**
This PR adds a new guide by the BP, AI/ML WGs and the Memory Safety SIG

### CODE REVIEW

File: docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+* **Secure Defaults & Configurations:** Include guidance such as: *“Prefer safe defaults in configurations – for example, use HTTPS by default, require strong encryption algorithms, and disable insecure protocols or options.”* By specifying this, the AI will be more likely to generate code that opts-in to security features. Always instruct the AI to follow the principle of least privilege (e.g. minimal file system permissions, least-privileged user accounts for services, etc.) in any configuration or code it proposes.  
+* **Testing for Security:** Encourage the AI to produce or suggest tests for critical code paths. In your instructions, add: *“When applicable, generate unit tests for security-critical functions (including negative tests to ensure the code fails safely).”* Automated testing is a recognized best practice – guides recommend including negative tests to verify that what *shouldn’t* happen, doesn’t happen.
+
```
Comment (balteravishay) [inline docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md:17]:
```suggestion
* **Call out and review stubbed code:** If the AI generates code that is stubbed or incomplete, instruct it to flag these areas for review. For example: *“If you generate placeholder code (e.g., `TODO` comments), ensure it is marked for security review before deployment.”* This will help ensure that any incomplete code does not inadvertently introduce vulnerabilities.

```

Comment (balteravishay) [reply docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md:17]: @mihaimaruseac is this what you had in mind for your suggestion?

File: docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# **Security-Focused Guide for AI Code Assistant Instructions**
+
+AI code assistants can significantly speed up development, but they need guidance to produce **secure** and robust code. This guide explains how to create custom instructions (e.g. a Copilot **instructions file**) that ensure the AI assistant accounts for application code security, supply chain safety, and platform or language-specific considerations and to embed a “security conscience” into the tool. In practice, this means fewer vulnerabilities making it into your codebase. Remember that these instructions should be kept concise, specific, and actionable – the goal is to influence the AI’s behavior without overwhelming it.
```
Comment (balteravishay) [inline docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md]:
```suggestion
AI code assistants can significantly speed up development, but they need guidance to produce **secure** and robust code. This guide explains how to create custom instructions (e.g. [GitHub Copilot instructions file](https://docs.github.com/en/copilot/how-tos/custom-instructions/adding-repository-custom-instructions-for-github-copilot), [Cline instructions file](https://docs.cline.bot/enterprise-solutions/custom-instructions), [Cursor rules](https://docs.cursor.com/context/rules), [Claude markdown](https://docs.anthropic.com/en/docs/claude-code/common-workflows#create-an-effective-claude-md-file), etc.) that ensure the AI assistant accounts for application code security, supply chain safety, and platform or language-specific considerations and to embed a “security conscience” into the tool. In practice, this means fewer vulnerabilities making it into your codebase. Remember that these instructions should be kept concise, specific, and actionable – the goal is to influence the AI’s behaviour without overwhelming it.
```

File: docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md (hunk 1 above)
Comment (mihaimaruseac) [reply docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md:17]: Yup!, thank you

**Comment by balteravishay** [comment]
Tracking change requests:
* Add TL/DR
* Add citations
* Add examples

**Comment by david-a-wheeler** [comment]
I'd like to see citations added that show that "this actually works", but don't make that a blocker, add them as we go.
